### PR TITLE
Adding Neighboring location to Victim's reported location.

### PR DESCRIPTION
**Pull request by Meghanaskanagal**
Adding Neighboring location to the Victim's display card when the status of the mission is either REPORTED or ASSIGNED.

Attaching the screenshot of the same.

Screenshot from 2019-12-27 17-40-49

### CODE REVIEW

File: src/Display.tsx [hunk 1]
```diff
@@ -109,31 +110,20 @@ const ShelterDetail: React.FC<ShelterDetailProps> = props => {
...
-    .then(jsonData => {
-      setShelterName(jsonData.map.shelter.map.name);
-    });
-  return (
-    <>
-      <GridItem span={3}>Designated Shelter: </GridItem>
-      <GridItem span={9}>{shelterName}</GridItem>
-    </>
-  );
-};
```
Comment (rottencandy) [inline src/Display.tsx]: Why is this removed?

File: src/Display.tsx [hunk 2]
```diff
@@ -144,7 +134,14 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
     .then(response => response.json())
     .then(jsonData => {
       if (jsonData.features.length) {
+        let neighbouringLocations = [];
+        for (let i = 1; i < 2; i++) {
+          neighbouringLocations.push(jsonData.features[i].place_name);
+        }
```
Comment (rottencandy) [inline src/Display.tsx]: This for loop runs for only one iteration.

File: src/Display.tsx [hunk 3]
```diff
@@ -171,22 +168,40 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
                 <FlexItem>Location:</FlexItem>
-                {props.data.status !== "REPORTED" && (
+                {props.data.status === status.reported && (
+                  <FlexItem>Neighbouring Location:</FlexItem>
+                )}
+                {props.data.status === status.assigned && (
+                  <FlexItem>Neighbouring Location:</FlexItem>
+                )}
+
+                {props.data.status !== status.reported && (
```
Comment (rottencandy) [inline src/Display.tsx]: It might be better to use inline if-else or ternary operator.
Something like: https://reactjs.org/docs/conditional-rendering.html

File: src/Display.tsx [hunk 4]
```diff
@@ -171,22 +168,40 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
+                )}
                 <FlexItem>{address}</FlexItem>
-                {props.data.status !== "REPORTED" && (
+                {props.data.status === status.assigned && (
+                  <FlexItem>{neighbourAddress}</FlexItem>
+                )}
+                {props.data.status === status.reported && (
+                  <FlexItem>{neighbourAddress}</FlexItem>
+                )}
+                {props.data.status !== status.reported && (
```
Comment (rottencandy) [inline src/Display.tsx]: Same here.

File: src/Display.tsx [hunk 5]
```diff
@@ -223,16 +223,19 @@ interface DisplayListProps {
...
   if (props.dataArray.length === 0) {
-    content = <p>No data available.</p>;
-  }
-  if (!props.isReady) {
-    content = <p>Loading...</p>;
+    if (props.isReady === false) {
+      content = <Alert variant="danger" isInline title="No data available" />;
+    } else {
+      content = null;
+    }
```
Comment (rottencandy) [inline src/Display.tsx]: `props.isReady` is false when the API has not yet returned a response.
The `content` var might not be displaying the correct message in this condition.

File: src/Display.tsx [hunk 6]
```diff
@@ -250,4 +253,4 @@ const DisplayList: React.FC<DisplayListProps> = props => {
   );
 };
 
-export default DisplayList;
+export default DisplayList;
```
Comment (rottencandy) [inline src/Display.tsx]: Add newline.

**Comment by rottencandy** [comment]
It might be better to have https://github.com/Emergency-Response-Demo/findmyrelative-frontend/pull/14/commits/a871404d3190838929545c43fc3746a70d1ae937 in separate PR.

File: src/Display.tsx (hunk 1 above)
Comment (Meghanaskanagal) [reply src/Display.tsx]: This was replaced by the FlexItems. And I had errors when cloned with map-display branch. The ShelterDetail was repeated twice. So, committed the latest code by removing the above code.

File: src/Display.tsx (hunk 2 above)
Comment (Meghanaskanagal) [reply src/Display.tsx]: It is because, We want only one neighbouring location.

If all the locations are required. Then we can use all the locations in the list.

Loop is just to print limited locations.

File: src/Display.tsx (hunk 3 above)
Comment (Meghanaskanagal) [reply src/Display.tsx]: Okay..

File: src/Display.tsx (hunk 4 above)
Comment (Meghanaskanagal) [reply src/Display.tsx]: Fine. Will do it.

File: src/Display.tsx (hunk 2 above)
Comment (rottencandy) [reply src/Display.tsx]: Okay, I think it is better to keep it simple for now and display only one location. Let's go with using index directly.
We will switch to using for loop if we decide to display multiple locations later.

File: src/Display.tsx [hunk 7]
```diff
@@ -218,39 +223,29 @@ const VictimDetail: React.FC<VictimDetailProps> = props => {
...
   }
-  if (!props.responseOk) {
-    content = (
-      <Alert
-        variant="danger"
-        isInline
-        title="Error: Emergency Response services unreachable"
-      />
-    );
-  }
```
[the remark this answers is not in the document]
Comment (Meghanaskanagal) [reply src/Display.tsx]: Because there is no proper response for unreachable service error from findmyrelative API.
Response is same for "no data" and 503 when we try to hit Emergency response demo api and when it's down.

Proper status response is needed in order to differentiate between findmyrelative service and emergency response demo service unreachable error.

Once API response comes as a text with status. We can un-comment it.

I have completed it prior so that dependency on the same won't come.
